Update GC cross-namespace note (#25091)
kubernetes/website: content/en/docs/concepts/workloads/controllers/garbage-collection.md

```

{{< note >}}
Cross-namespace owner references are disallowed by design.

Namespaced dependents can specify cluster-scoped or namespaced owners.
A namespaced owner **must** exist in the same namespace as the dependent.
If it does not, the owner reference is treated as absent, and the dependent
is subject to deletion once all owners are verified absent.

Cluster-scoped dependents can only specify cluster-scoped owners.
In v1.20+, if a cluster-scoped dependent specifies a namespaced kind as an owner,

> it is treated as having an unresolveable owner reference, and is not able to be garbage collected.

Reviewer: so in this case also an event is generated as per the next statement below ?

Author: yes

In v1.20+, if the garbage collector detects an invalid cross-namespace

> or cluster-scoped -> namespaced `ownerReference`, it reports a warning Event 

Reviewer: nit: `or cluster-scoped dependent referencing a namespaced ownerr` or whatever is better than using the arrow ?

> or cluster-scoped -> namespaced `ownerReference`, it reports a warning Event 

Reviewer 2: nit:
```suggestion
or cluster-scoped → namespaced `ownerReference`, it reports a warning Event 
```

> with a reason of `OwnerRefInvalidNamespace` with an `involvedObject` of the invalid dependent.

Reviewer 2: ```suggestion
with a reason of `OwnerRefInvalidNamespace` and an `involvedObject` of the invalid dependent.
```

> These events can be viewed by running `kubectl get events -A --field-selector=reason=OwnerRefInvalidNamespace`

Reviewer 2: ```suggestion
You can check for that kind of Event by running
`kubectl get events -A --field-selector=reason=OwnerRefInvalidNamespace`.
```

```

{{< note >}}

> Cross-namespace owner references are disallowed by design.

Reviewer 3: nit: What is an "owner reference"?

Author: ownerReferences are described in the paragraphs above

Reviewer 3: OK. It may help to break down the two cases (if I understand correctly). Does line 64 apply to the following paragraph or both pargraphs?
Is this true:
Dependent objects are only owned (referenced) by objects within their namespace or cluster scope.

There are two types of scoping that affect object ownership and garbage collection:

1. Namespaced
Namespaced owners **must** exist in the same namespace as the dependent.
...
1. Cluster-scoped
Cluster-scoped dependents can only specify cluster-scoped owners.
...

Reviewer 4: Hi @liggitt after this comment is addressed, I think this would be good to merge. PTAL

Reviewer 3: Hi. I think the text is okay to merge. In the future, the content could be updated --  make the text easier to digest and localize.
/lgtm

Namespaced dependents can specify cluster-scoped or namespaced owners.
A namespaced owner **must** exist in the same namespace as the dependent.
If it does not, the owner reference is treated as absent, and the dependent
is subject to deletion once all owners are verified absent.

Cluster-scoped dependents can only specify cluster-scoped owners.
In v1.20+, if a cluster-scoped dependent specifies a namespaced kind as an owner,
it is treated as having an unresolveable owner reference, and is not able to be garbage collected.

In v1.20+, if the garbage collector detects an invalid cross-namespace `ownerReference`,
or a cluster-scoped dependent with an `ownerReference` referencing a namespaced kind, a warning Event 
with a reason of `OwnerRefInvalidNamespace` and an `involvedObject` of the invalid dependent is reported.
You can check for that kind of Event by running

> `kubectl get events -A --field-selector=reason=OwnerRefInvalidNamespace`

Reviewer 2: attonit:
```suggestion
`kubectl get events -A --field-selector=reason=OwnerRefInvalidNamespace`.
```

Author: done
